### PR TITLE
Fix block not on mesh error

### LINKED ISSUE #30610: Connected BlockID is of size zero for pre-declared boundaries
### Bug Description

When constucting the boundaries of blocks, sidesets, and nodesets not directly associated with the mesh undewr the `Mesh` header, setting up the initial conditions within Pressure.C do not properly account for block_ids.size()=0.

### Steps to Reproduce

A simulation with material blocks and pressure boundary conditions, but with blocks that do not yet exist on the mesh will give an error.  Experienced when dealing with `MeshModifiers` under work with new uses for `SidesetsAroundSubdomainUpdater` and physics involving the `Pressure` class.

### Impact

Resolve problem for blocks not physically on the mesh, no changes to existing code.

### [Optional] Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by vanwdani**
Closes #30610

## Reason
Fix bug issue #30610

## Design
Check size of block_ids created outside of physical mesh

## Impact
Resolve bug

### CODE REVIEW

File: modules/solid_mechanics/src/bcs/Pressure.C [hunk 1]
```diff
@@ -89,7 +89,10 @@ PressureTempl<is_ad>::initialSetup()
     block_ids.insert(bids.begin(), bids.end());
   }
 
-  _coord_type = _fe_problem.getCoordSystem(*block_ids.begin());
+  if (block_ids.size())
+    _coord_type = _fe_problem.getCoordSystem(*block_ids.begin());
```
Comment (lindsayad) [inline modules/solid_mechanics/src/bcs/Pressure.C:93]: it seems like in debug mode we should check whether the coord system is the same for all specified blocks

**Review by lindsayad — DISMISSED** [review]

File: modules/solid_mechanics/src/bcs/Pressure.C [hunk 2]
```diff
@@ -89,7 +89,10 @@ PressureTempl<is_ad>::initialSetup()
...
+  else
+    _coord_type = _fe_problem.mesh().getUniqueCoordSystem();
```
Comment (GiudGiud) [inline modules/solid_mechanics/src/bcs/Pressure.C:95]: and in this case, we do check in opt mode
```
Moose::CoordinateSystemType
MooseMesh::getUniqueCoordSystem() const
{
  const auto unique_system = _coord_sys.find(*meshSubdomains().begin())->second;
  // Check that it is actually unique
  bool result = std::all_of(
      std::next(_coord_sys.begin()),
      _coord_sys.end(),
      [unique_system](
          typename std::unordered_map<SubdomainID, Moose::CoordinateSystemType>::const_reference
              item) { return (item.second == unique_system); });
  if (!result)
    mooseError("The unique coordinate system of the mesh was requested by the mesh contains "
               "multiple blocks with different coordinate systems");

  if (usingGeneralAxisymmetricCoordAxes())
    mooseError("General axisymmetric coordinate axes are being used, and it is currently "
               "conservatively assumed that in this case there is no unique coordinate system.");

  return unique_system;
}
```

**Review by GiudGiud — COMMENTED** [review]
needs a test to prevent regressing

File: modules/solid_mechanics/src/bcs/Pressure.C (hunk 1 above)
Comment (lindsayad) [reply modules/solid_mechanics/src/bcs/Pressure.C:93]: This class certainly does not work if there are coordinate systems that are not matching. As is done in Navier-Stokes there are potentially additional pressure terms in RZ. They could change the class to not store a single `_coord_type` but until then there needs to be an error check. What I failed to realize is that if I expand the file the error check is already right below

File: modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i [hunk 3]
```diff
@@ -0,0 +1,134 @@
+#
+#
+#
+
```
Comment (GiudGiud) [inline modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i]:
```suggestion
```

File: modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    ymin = 0
+    ymax = 1
+  []
+  [rename]
+    type = RenameBoundaryGenerator
+    input = MeshGenerator
+    old_boundary = 'right top'
+    new_boundary = 'surround surround'
+  []
+  final_generator = rename
```
Comment (GiudGiud) [inline modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i]:
```suggestion
```

File: modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  [disp_x]
+  []
+  [disp_y]
+  []
+[]
+
+[Kernels]
+  [SolidMechanics]
+  []
+[]
```
Comment (GiudGiud) [inline modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i:54]:
```suggestion
[Physics]
  [SolidMechanics]
    [QuasiStatic]
    []
  []
[]
```

File: modules/solid_mechanics/src/bcs/PressureBase.C [hunk 6]
```diff
@@ -79,7 +79,13 @@ PressureBaseTempl<is_ad>::initialSetup()
     block_ids.insert(bids.begin(), bids.end());
   }
 
-  _coord_type = _fe_problem.getCoordSystem(*block_ids.begin());
+  if (block_ids.size())
+    _coord_type = _fe_problem.getCoordSystem(*block_ids.begin());
+  else
+  {
+    mooseWarning("Getting the coordinates from the mesh.");
+    _coord_type = _fe_problem.mesh().getUniqueCoordSystem();
```
Comment (GiudGiud) [inline modules/solid_mechanics/src/bcs/PressureBase.C]:
```suggestion
  {
    mooseInfo("No connected blocks were found, the coordinate system type is obtained from the mesh.");
    _coord_type = _fe_problem.mesh().getUniqueCoordSystem();
```

File: modules/solid_mechanics/test/tests/pressure/tests [hunk 7]
```diff
@@ -63,4 +63,22 @@
...
+  []
+    [predeclared_boundary_warning]
+    type = RunException
+    input = 'pressure_predeclared_boundaries.i'
+    issues = '#30611'
+    design = 'syntax/BCs/Pressure/index.md'
+    allow_warnings = false
+    expect_err = "Getting the coordinates from the mesh."
+    requirement = 'The Pressure boundary condition action shall return an expected warning/error when the predecleared boundary is used for getting the coordinates from the mesh.'
+  []
```
Comment (GiudGiud) [inline modules/solid_mechanics/test/tests/pressure/tests]:
```suggestion
  [predeclared_boundary_warning]
    type = RunApp
    input = 'pressure_predeclared_boundaries.i'
    issues = '#30611'
    design = 'syntax/BCs/Pressure/index.md'
    expect_out = "No connected blocks were found, the coordinate system is obtained from the mesh."
    requirement = 'The Pressure boundary condition action shall inform the user when retrieving the coordinate system type from the mesh.'
  []
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]

File: modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+  []
+[]
+
```
Comment (GiudGiud) [inline modules/solid_mechanics/test/tests/pressure/pressure_predeclared_boundaries.i:97]:
```suggestion

[Problem]
  kernel_coverage_check = false
[]

```

**Review by GiudGiud — APPROVED** [review]

**Review by GiudGiud — CHANGES_REQUESTED** [review]
test is not converging

**Review by GiudGiud — APPROVED** [review]
VTB failure is unrelated
